csrc/ir/internal_base_nodes.h

```diff
@@ -683,6 +693,7 @@ class TensorDomain : public Val {
   const std::vector<IterDomain*> logical_domain_;
   std::vector<IterDomain*> allocation_domain_;
   std::vector<IterDomain*> loop_domain_;
+  std::vector<IterDomain*> additional_ids_;
```

Comment:
I believe this mechanism can be used for fixing:
https://github.com/NVIDIA/Fuser/pull/2379#discussion_r1653381767
But I will leave it as a future work.